File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        for value in nums2:nums1.append(value)
        nums1 = sorted(nums1)
        if len(nums1)%2 == 0:
            left = 0
            right = len(nums1)-1
            while left != right-1:
                left+=1
                right-=1
            return((nums1[left]+nums1[right])/2)
        else:
            left = 0
            right = len(nums1)-1
            while left != right:
                left+=1
                right-=1
            return(nums1[left])

File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.go
package main

import (
	"fmt"
	"sort"
)

func findMedianSortedArrays(nums1 []int, nums2 []int)  float64{
    nums1 = append(nums1, nums2...)
	sort.Ints(nums1)
	if len(nums1)%2 == 0 {
		left := 0
		right := len(nums1)-1
		for{
			if left == right-1 {
				break
			}
			left+=1
			right-=1
		}
		b := float64(nums1[left])+float64(nums1[right])
		
		return float64(b/2)
	}else{
		left := 0
		right := len(nums1)-1
		for{
			if left == right {
				break
			}
			left+=1
			right-=1
		}
		return float64(nums1[left])
	}
	
}